lib/components/grpc.ts

```diff
@@ -463,6 +474,9 @@ function getServiceInstanceReflectCached(
 
     const service = getServiceInstanceReflect(config, endpointData, grpcOptions, credentials);
     _.set(reflectionServiceInstancesMap, cacheKey, service);
+    service.catch(() => {
+        _.set(reflectionServiceInstancesMap, cacheKey, undefined);
```

Comment:
1. Let's add logs here to be able to find such things in logs
2. I think it is better to re-throw initial error after clearing a cache, otherwise the function above `const service = await getService()` will receive `service === undefined`

Reply:
1. Good point
2. Re-throw where? If we do so, it will lead to exactly the same problem as we had with the refreshCache. The `getService()` will not return the `undefined`, it'll return a [rejected promise](https://github.com/gravity-ui/gateway/pull/34/files/9fd0d9f07deee7b535e7ee3357aceadb33017dc6#diff-a8f41f4692e70dce3fb4ada1729584eefb291749c963779360cd15eb1ecb6542R475) in this case. Again, the `getServiceInstanceReflectCached` function itself doesn't handle promise here, it just return it two lines below. But the promise itself, in case of an exception, will execute this function passed to its catch method.

Reply:
Well, by the way, it's not that easy to do good logging via `Context.log` here, because a context might be already ended by the moment we catch an exception. Because the request the context belongs to might be handled already.